api/v3/Case/Getdetails.php

```diff
@@ -88,6 +94,10 @@ function civicrm_api3_case_getdetails(array $params) {
     CRM_Civicase_APIHelpers_CasesByContactInvolved::filter($sql, $params['contact_involved']);
   }
 
+  if (!empty($params['exclude_for_client_id'])) {
+    $sql->where("a.id NOT IN (SELECT case_id FROM civicrm_case_contact WHERE contact_id = {$params['exclude_for_client_id']})");
```

Comment:
not 100% sure if this is SQL Injection safe, but better make sure we are safe by doing this instead:

```php
$sql->where('a.id NOT IN (SELECT case_id FROM civicrm_case_contact WHERE contact_id = #contact_id', [
  '#contact_id' => $params['exclude_for_client_id'],
]);
```